xcodeproj/internal/bazel_integration_files/copy_dsyms.sh

```diff
@@ -2,6 +2,47 @@
 
 set -euo pipefail
 
+# This is for testing, removed before merging
+set -x
+exec > >(tee -i ~/logfile.txt)
+exec 2>&1
+#############################################
+
+function patch_dsym() {
+  readonly dsym_name="$1"
+  readonly binary_name="${dsym_name%%.*}"
+  readonly binary_path="${dsym_name}/Contents/Resources/DWARF/${binary_name}"
+  if [[ ! -f "$binary_path" ]]; then
+    echo "dSYM DWARF ${binary_path} does not exist." \
+    "Skip dSYM patch."
+    return 1
+  fi
+  dwarf_uuid=$(dwarfdump --uuid "${binary_path}" | cut -d ' ' -f 2)
+  readonly dwarf_uuid
+  if [[ -z "${dwarf_uuid// /}" ]]; then
+    echo "Failed to get dSYM uuid." \
+    "Skip dSYM patch."
+    return 1
+  fi
+  readonly plist_path="${dsym_name}/Contents/Resources/${dwarf_uuid}.plist"
+  readonly execution_root="$PROJECT_DIR"
+  readonly workspace_root="$BAZEL_WORKSPACE_ROOT"
+  cat > "${plist_path}" <<EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+  <dict>
+    <key>DBGVersion</key>
+    <string>3</string>
+    <key>DBGSourcePathRemapping</key>
+    <dict>
+      <key>$execution_root</key>
+      <string>$workspace_root</string>
```

Comment:
The remapping should be the same as in `.lldbinit`, changing relative paths to absolute. In particular, it should be these three:

- `./bazel-out/` -> `$BAZEL_OUT`
- `./external/` -> `$BAZEL_EXTERNAL`
- `./` -> `$SRCROOT`

Might also need to do the same with `RESOLVED_REPOSITORIES` (and if so, might need to do the ordering trick as well).

Reply:
Note:
I think there is still something wrong with the path, the Instruments are complaining about finding the source.
Will investigate this.